# HG changeset patch
# Parent  3b606e677ad418a8b63713488eb7d52ec0368c33
Fix how the program name is displayed in the help message

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@
 
 SET(CPACK_PACKAGE_VERSION_MAJOR "0")
 SET(CPACK_PACKAGE_VERSION_MINOR "9")
-SET(CPACK_PACKAGE_VERSION_PATCH "3")
+SET(CPACK_PACKAGE_VERSION_PATCH "4")
 SET(LIGHTSD_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
 
 MESSAGE(STATUS "CMake version: ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}")
@@ -70,7 +70,7 @@
 IF (CMAKE_BUILD_TYPE MATCHES "DEBUG")
     ADD_DEFINITIONS("-DQUEUE_MACRO_DEBUG=1")
     IF (CMAKE_COMPILER_IS_GNUCC)
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g3 -ggdb")
+        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g3 -ggdb3")
     ENDIF ()
 ENDIF ()
 
diff --git a/core/lightsd.c b/core/lightsd.c
--- a/core/lightsd.c
+++ b/core/lightsd.c
@@ -160,6 +160,9 @@
 int
 main(int argc, char *argv[], char *envp[])
 {
+    char progname[32];
+    memcpy(progname, argv[0], LGTD_MIN(sizeof(progname), strlen(argv[0])));
+
     lgtd_daemon_setup_proctitle(argc, argv, envp);
 
     lgtd_configure_libevent();
@@ -184,7 +187,7 @@
     const char short_opts[] = "l:c:s:fdu:g:thv:V";
 
     if (argc == 1) {
-        lgtd_usage(argv[0]);
+        lgtd_usage(progname);
     }
 
     for (int rv = getopt_long(argc, argv, short_opts, long_opts, NULL);
@@ -195,7 +198,7 @@
             (void)0;
             char *sep = strrchr(optarg, ':');
             if (!sep || !sep[1]) {
-                lgtd_usage(argv[0]);
+                lgtd_usage(progname);
             }
             *sep = '\0';
             if (!lgtd_listen_open(optarg, sep + 1)) {
@@ -227,7 +230,7 @@
             lgtd_opts.log_timestamps = false;
             break;
         case 'h':
-            lgtd_usage(argv[0]);
+            lgtd_usage(progname);
         case 'v':
             for (int i = 0;;) {
                 const char *verbose_levels[] = {
@@ -243,7 +246,7 @@
             }
             break;
         case 'V':
-            printf("lightsd %s\n", LGTD_VERSION);
+            printf("%s %s\n", progname, LGTD_VERSION);
             lgtd_cleanup();
             return 0;
         case 'p':
@@ -261,7 +264,7 @@
             );
             return 0;
         default:
-            lgtd_usage(argv[0]);
+            lgtd_usage(progname);
         }
     }
 
